src/Features/CSharp/Portable/CodeFixes/PopulateSwitch/PopulateSwitchCodeFixProvider.cs

```diff
@@ -0,0 +1,196 @@
+
+            var containsDefaultLabel = false;
+
+            var caseLabels = new List<ExpressionSyntax>();
+            foreach (var section in switchBlock.Sections)
+            {
+                foreach (var label in section.Labels)
+                {
+                    var caseLabel = label as CaseSwitchLabelSyntax;
+                    if (caseLabel != null)
+                    {
+                        caseLabels.Add(caseLabel.Value);
+                        continue;
+                    }
+
+                    if (label.IsKind(SyntaxKind.DefaultSwitchLabel))
+                    {
+                        containsDefaultLabel = true;
+                    }
+                }
+            }
+
+            var missingLabels = GetMissingLabels(model, caseLabels, enumType);
+
+            var breakStatement = SyntaxFactory.BreakStatement();
+            var statements = SyntaxFactory.List(new List<StatementSyntax> { breakStatement });
+
+            var newSections = SyntaxFactory.List(switchBlock.Sections);
+            foreach (var label in missingLabels)
+            {
+                // If an existing simplified label exists, it means we can assume that works already and do it ourselves as well (ergo: there is a static using)
+                var caseLabel =
+                    SyntaxFactory.CaseSwitchLabel(
+                        SyntaxFactory.ParseExpression($"{fullyQualifiedEnumType}.{label}")
+                                     .WithTrailingTrivia(SyntaxFactory.ParseTrailingTrivia(Environment.NewLine)));
+
+                var section =
+                    SyntaxFactory.SwitchSection(SyntaxFactory.List(new List<SwitchLabelSyntax> { caseLabel }), statements)
+                                 .WithAdditionalAnnotations(Formatter.Annotation);
+
+                // ensure that the new cases are above the last section if a default case exists, but below all other sections
+                if (containsDefaultLabel)
+                {
+                    // this will not give an index-out-of-bounds error because we know there is at least one case block
+                    newSections = newSections.Insert(switchBlock.Sections.Count - 1, section);
+                }
+                else
+                {
+                    newSections = newSections.Add(section);
+                }
+            }
+            
+            if (!containsDefaultLabel)
+            {
+                newSections = newSections.Add(SyntaxFactory.SwitchSection(SyntaxFactory.List(
+                        new List<SwitchLabelSyntax> { SyntaxFactory.DefaultSwitchLabel() }), statements));
+            }
+
+            var newNode = switchBlock.WithSections(newSections).WithAdditionalAnnotations(Formatter.Annotation, Simplifier.Annotation);
+
+            var newRoot = root.ReplaceNode(switchBlock, newNode);
+            return await Simplifier.ReduceAsync(document.WithSyntaxRoot(newRoot)).ConfigureAwait(false);
```

Comment:
I don't think you need to call `ReduceAsync` explicitly here. In line 125, you added a `Simplifier.Annotation` which is sufficient and the simplifier will eventually perform the required operations.